python/ray/_private/runtime_env/conda.py

```diff
@@ -306,28 +307,40 @@ async def create(
         context: RuntimeEnvContext,
         logger: Optional[logging.Logger] = default_logger,
     ) -> int:
-        logger.debug("Setting up conda for runtime_env: " f"{runtime_env.serialize()}")
-        protocol, hash = parse_uri(uri)
-        conda_env_name = self._get_path_from_hash(hash)
-
-        conda_dict = _get_conda_dict_with_ray_inserted(runtime_env, logger=logger)
+        # Currently create method is still a sync process, to avoid blocking
+        # the loop, need to run this function in another thread.
+        # TODO(Catch-Bull): Refactor method create into an async process, and
+        # make this method running in current loop.
```

Comment:
> Thanks @Catch-Bull. Could you please also leave a comment for why we're doing this and a TODO to remove it once refactored?

@edoakes I add annotations and TODO, and copy those to conda.py, py_modules.py, and working_dir.py